### ISSUE TITLE
Updating from pre 7.3.0 to 7.3.0+ fails on OpenStack

### ISSUE THREAD

**Issue by bloeys** [cloudfoundry/bosh-bootloader#462]
Hello,

So the increased automation with OS starting with v7.3.0 is pretty awesome, however, it doesn't seem that we can upgrade from an earlier version before this automation was put into place.

bbl will not use the old settings and instead create a new network, router etc... and try to place the VMs there, which isn't the behavior I would expect.

Additionally, even then it would fail because it would assign a new floating IP to the Jumpbox but the certificate will still use the old IP, and so connection is refused and so bbl up fails.

Is there currently support for upgrading from these previous versions on OS?

**Comment by nmahoney-pivotal** [comment]
Hi @bloeys,

We pulled the OS support in as a [PR](https://github.com/cloudfoundry/bosh-bootloader/pull/431), and unfortunately I don't think we tested the upgrade case. We probably should have made this a major version bump.

How did this affect you? Did the attempted upgrade break your environment? If so, were you able to recover it?

**Comment by bloeys** [comment]
Hello @nmahoney-pivotal,

Well it **almost** broke the environment. We already had a test environment with several deployments setup using the old way (manually created networks, keys etc...), but what the upgrade did is that it ignored all these and acted as if it is a new deployment.

It deleted the original Jumpbox and tried to recreate it with the new release/stemcell/IP, however the recreation of the Jumpbox failed since the new Jumpbox was assigned the new IP but the certificate was outdated and so it couldn't connect and verify the Jumpbox is up so bbl up failed. As a consequence, the bosh VM was (luckily) unaffected, but of course bosh was unusable at that point.

I can't recall the exact steps I took to restore everything, but in general (after reverting to an older bbl version and deleting the new Jumpbox) I had to remove some details regarding the Jumpbox from the vars folder, like the stemcell info, which I think was from the `terraform.tfstate` and `jumpbox-state.json` files.

This forced bbl to recreate the Jumpbox but with the information from the old version and the old IP, so everything was restored. I hope this provides an explanation of what happened and why things actually failed.

To have a proper update process bbl will have to use the existing deployment information if available and not create anything on OpenStack, otherwise existing deployments will be lost to bosh. If there is no 'vars' file then I think bbl can go ahead with setting up the infrastructure on OS, but otherwise simply reuse the existing data.
